recirq/qcqmc/trial_wf.py

```diff
@@ -0,0 +1,1584 @@
+    for perm in itertools.permutations(initial_bitstring):
+        spin_sector_bitstrings.add(perm)
+
+    for bitstring_a, bitstring_b in itertools.product(spin_sector_bitstrings, repeat=2):
+        yield bitstring_a + bitstring_b
+
+
+def _get_fermion_qubit_map_pp_plus(*, n_qubits: int) -> Dict[int, cirq.GridQubit]:
+    if n_qubits == 4:
+        return get_4_qubit_fermion_qubit_map()
+    elif n_qubits == 8:
+        return get_8_qubit_fermion_qubit_map()
+    elif n_qubits == 12:
+        return get_12_qubit_fermion_qubit_map()
+    elif n_qubits == 16:
+        return get_16_qubit_fermion_qubit_map()
+    else:
+        raise NotImplementedError()
+
+
+def _get_mode_qubit_map_pp_plus(
+    *, n_qubits: int
+) -> Dict[FermionicMode, cirq.GridQubit]:
+    """A map from Fermionic modes to qubits for our particular circuits.
+
+    This function dispatches to _get_fermion_qubit_map_pp_plus, and ultimately
+    to get_X_qubit_fermion_qubit_map for specific values of X but it translates
+    this logic to a new system that uses the FermionicMode class rather than
+    opaque combinations of integers and strings.
+    """
+    old_fermion_qubit_map = _get_fermion_qubit_map_pp_plus(n_qubits=n_qubits)
+
+    n_orb = n_qubits // 2
+
+    mode_qubit_map = {}
+
+    for i in range(n_orb):
+        mode_qubit_map[FermionicMode(i, "a")] = old_fermion_qubit_map[2 * i]
+        mode_qubit_map[FermionicMode(i, "b")] = old_fermion_qubit_map[2 * i + 1]
+
+    return mode_qubit_map
+
+
+def _get_reorder_func(
+    *,
+    mode_qubit_map: Mapping[FermionicMode, cirq.Qid],
+    ordered_qubits: Sequence[cirq.Qid],
+) -> Callable[[int, int], int]:
+    """This is a helper function that allows us to reorder fermionic modes.
+
+    Under the Jordan-Wigner transform, each fermionic mode is assigned to a
+    qubit. If we are provided an openfermion FermionOperator with the modes
+    assigned to qubits as described by mode_qubit_map this function gives us a
+    reorder_func that we can use to reorder the modes (with
+    openfermion.reorder(...)) so that they match the order of the qubits in
+    ordered_qubits. This is necessary to make a correspondence between
+    fermionic operators / wavefunctions and their qubit counterparts.
+
+    Args:
+        mode_qubit_map: A dict that shows how each FermionicMode is mapped to a qubit.
+
+        ordered_qubits: An ordered sequence of qubits.
+    """
+    qubits = list(mode_qubit_map.values())
+    assert len(qubits) == len(ordered_qubits)
+
+    # We sort the key: value pairs by the order of the values (qubits) in
+    # ordered_qubits.
+    sorted_mapping = list(mode_qubit_map.items())
+    sorted_mapping.sort(key=lambda x: ordered_qubits.index(x[1]))
+
+    remapping_map = {}
+    for i, (mode, _) in enumerate(sorted_mapping):
+        openfermion_index = 2 * mode.orb_ind + (0 if mode.spin == "a" else 1)
+        remapping_map[openfermion_index] = i
+
+    print("remapping_map:")
```

Comment:
Is this debugging info?  Should we keep or remove these print statements?

Reply:
Removed